pkg/apis/core/validation/shoot.go

```diff
@@ -997,7 +997,8 @@ func validateKubernetesVersionUpdate125(new, old *core.Shoot) field.ErrorList {
 		allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "kubernetes", "version"), old.Spec.Kubernetes.Version, "Invalid old kubernetes version"))
 	}
 
-	if newShootVersionGreaterEqual125 && oldShootVersionLess125 {
+	// if the shoot is workerless, continue. we have disabled the policy/v1beta1/podsecuritypolicies API already.
```

Comment:
```suggestion
	// since we have disabled the policy/v1beta1/podsecuritypolicies API for workerless shoots, we only need to check for disabled PSPs for regular shoots.
```